### PR TITLE
OTEL Processing Rules for source template

### PR DESCRIPTION
**Pull request by sumoanema**
…ate are in beta release). 

## Purpose of this pull request

This include details for include and exclude filter processing rules for logs and metrics which can be used for OTRM source template (Beta released) using otel collector.

## Select the type of change

- [ ] Minor Changes - Typos, formatting, slight revisions
- [ ] Update Content - Revisions, updating sections
- [X] New Content - New features, sections, pages, tutorials
- [ ] Site and Tools - .clabot, version updates, maintenance, dependencies, new packages for the site (Docusaurus, Gatsby, React, etc.)

## Ticket (if applicable)
https://sumologic.atlassian.net/browse/SUMO-231512
https://sumologic.atlassian.net/browse/SUMO-239785

### CODE REVIEW

File: docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules-otel.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+id: include-and-exclude-rules-otel
+title: Include and Exclude Rules OpenTelemetry (Beta)
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules-otel.md:3]:
```suggestion
title: Include and Exclude Rules for OpenTelemetry (Beta)
```

File: docs/send-data/opentelemetry-collector/processing-rules/metrics-include-and-exclude-rules-otel.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+id: metrics-include-and-exclude-rules-otel
+title: Metrics Include and Exclude Rules OpenTelemetry (Beta)
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/metrics-include-and-exclude-rules-otel.md:3]:
```suggestion
title: Metrics Include and Exclude Rules for OpenTelemetry (Beta)
```

File: docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules-otel.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* An exclude rule functions as a denylist filter where the matching data is not sent to Sumo Logic.
+* An include rule functions as an allowlist filter where only matching data is sent to Sumo Logic.
+
+As a best practice, specify these rules to match the lesser volume of data.
+
+* If you want to **collect the majority of data** from a source template, provide **exclude** rules to match (filter out) the lesser volume of data.
+* If you want to **collect a small set of data** from a source template, provide **include** rules to match (filter in) the lesser volume of data.
+
+For example, to include only messages coming from a Windows Event log with id 8015,
+you can add a Logs Filter to source template and select the "Type" of the filter as "Include message that match" and can use the following filter regular expression:
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules-otel.md:25]:
```suggestion
For example, to include only messages coming from a Windows Event log with ID `8015`, you can add a Logs Filter to the source template and select the **Type** of the filter as "Include message that match", and can use the following filter regular expression:
```

File: docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules-otel.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+```
+.*"id":8015.*
+```
+
+<img src={useBaseUrl('img/send-data/opentelemetry-collector/processingrule-include-logs.png')} alt="collector-installation-completion-page" style={{border:'1px solid gray'}} width="700" />
+
+## Rules and Limitations
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules-otel.md:33]:
```suggestion
## Rules and limitations
```

File: docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules-otel.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+When writing regular expression rules, you must follow these rules:
+
+* Your rule must be [RE2 compliant](https://github.com/google/re2/wiki/Syntax).
+* If your rule matches **only a section**, of the log line the full log line will be matched.
+* For **single line messages**, its not mandatory to prefix and suffix the regex expression with .\*
+* Exclude rules take priority over include rules. Include rules are processed first, however, if an exclude rule matches data that matched the include rule filter, the data is excluded.
+* If two or more rules are listed, the assumed Boolean operator is OR.
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules-otel.md:41]:
```suggestion
* If your rule matches *only a section* of the log line, the full log line will be matched.
* For *single line messages*, it is not mandatory to prefix and suffix the regex expression with `.\*`.
* Exclude rules take priority over include rules. Include rules are processed first. However, if an exclude rule matches data that matched the include rule filter, the data is excluded.
* If two or more rules are listed, the assumed Boolean operator is `OR`.
```

File: docs/send-data/opentelemetry-collector/processing-rules/index.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+---
+title: Processing Rules for OpenTelemetry (Beta)
+description: Use Sumo Logic processing rules for opentelemetry agent with OTRM source template
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/index.md:3]:
```suggestion
description: Use Sumo Logic processing rules for an OpenTelemetry agent with an OpenTelemetry remote management (OTRM) source template.
```

File: docs/send-data/opentelemetry-collector/processing-rules/index.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+---
+<head>
+  <meta name="robots" content="noindex" />
+</head>
+
+<p><a href="/docs/beta"><span className="beta">Beta</span></a></p>
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+Processing rules can be used with OpenTelemetry collector for different source template in OTRM (Open telemetry remote management). These processing rules can filter and can mask the data sent to Sumo Logic from a OpenTelemtry collector which is remotely managed by Sumo Logic. The rules affect only the data sent to Sumo Logic; logs and metrics on your end remain intact and unchanged. Data filtered by a OpenTelemetry collector using Processing Rules does not count towards your daily data volume quota.
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/index.md:13]:
```suggestion
Processing rules can be used with an OpenTelemetry collector for different source templates in OTRM (OpenTelemetry remote management). These processing rules can filter and can mask the data sent to Sumo Logic from an OpenTelemetry collector which is remotely managed by Sumo Logic. The rules affect only the data sent to Sumo Logic; logs and metrics on your end remain intact and unchanged. Data filtered by an OpenTelemetry collector using processing rules does not count towards your daily data volume quota.
```

File: docs/send-data/opentelemetry-collector/processing-rules/index.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+**Processing Rules for Logs Collection** support the following rule types:
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/index.md:15]:
```suggestion
Processing rules for logs collection support the following rule types:
```

File: docs/send-data/opentelemetry-collector/processing-rules/index.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+**Processing Rules for Metrics Collection** support the following rule types:
+
+* [Exclude metrics that match](metrics-include-and-exclude-rules-otel.md). Remove metrics that you do not want to send to Sumo Logic at all ("denylist" filter).
+* [Include metrics that match](metrics-include-and-exclude-rules-otel.md). Send only selected metrics to your Sumo Logic account (an "allowlist" filter). 
+
+## Limitations
+
+* Regular expressions must be [RE2 compliant](https://github.com/google/re2/wiki/Syntax).
+
+## How do Processing Rules Work Together?
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/index.md:29]:
```suggestion
## How do processing rules work together?
```

File: docs/send-data/opentelemetry-collector/processing-rules/metrics-include-and-exclude-rules-otel.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+description: You can use metrics processing rules to specify what metrics are send to Sumo Logic using OpenTelemetry Collector.
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/metrics-include-and-exclude-rules-otel.md:4]:
```suggestion
description: You can use metrics processing rules to specify what metrics are sent to Sumo Logic using OpenTelemetry Collector.
```

File: docs/send-data/opentelemetry-collector/processing-rules/metrics-include-and-exclude-rules-otel.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+* An exclude rule functions as a denylist filter where all data is sent except matching data to Sumo Logic.
+* An include rule functions as an allowlist filter where only matching data is sent to Sumo Logic.
+
+As a best practice, specify these rules to match the lesser volume of data.
+
+* If you want to collect the majority of data from a source template, provide exclude rules to match (filter out) the lesser volume of data
+* If you want to collect a small set of data from a source template, provide include rules to match (filter in) the lesser volume of data.
+
+## Metric Filter Example
```
Comment (jpipkin1) [inline docs/send-data/opentelemetry-collector/processing-rules/metrics-include-and-exclude-rules-otel.md:24]:
```suggestion
## Metric filter example
```

**Review by jpipkin1 — COMMENTED** [review]
@sumoanema, I have a few suggestions. 

I also went ahead and made some updates myself (see commits "[Update metric filter examples section](https://github.com/SumoLogic/sumologic-documentation/pull/4015/commits/1f9ef34dc2f490cf3db7f17b7c869bae5de96301)" and "[Add cards to index page](https://github.com/SumoLogic/sumologic-documentation/pull/4015/commits/bab93700296b2e6b2d7608bda9f8749debc20381)"). 

Let me know if you have any questions.

**Review by jpipkin1 — APPROVED** [review]

File: docs/send-data/opentelemetry-collector/processing-rules/index.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+<div className="box-wrapper" >
+<div className="box smallbox card">
+  <div className="container">
+  <a href="/docs/send-data/opentelemetry-collector/processing-rules/include-and-exclude-rules"><img src={useBaseUrl('img/icons/operations/rules.png')} alt="icon" width="40"/><h4>Include and Exclude Rules for OpenTelemetry </h4></a>
+  <p>Use include and exclude processing rules to specify what kind of data is sent to Sumo Logic using OpenTelemetry Collector.</p>
+  </div>
+</div>
+<div className="box smallbox card">
+  <div className="container">
+  <a href="/docs/send-data/opentelemetry-collector/processing-rules/metrics-include-and-exclude-rules-otel"><img src={useBaseUrl('img/icons/operations/rules.png')} alt="icon" width="40"/><h4>Metrics Include and Exclude Rules for OpenTelemetry</h4></a>
```
Comment (kimsauce) [inline docs/send-data/opentelemetry-collector/processing-rules/index.md:51]:
```suggestion
  <a href="/docs/send-data/opentelemetry-collector/processing-rules/metrics-include-and-exclude-rules"><img src={useBaseUrl('img/icons/operations/rules.png')} alt="icon" width="40"/><h4>Metrics Include and Exclude Rules for OpenTelemetry</h4></a>
```

**Comment by jpipkin1** [comment]
@kimsauce, my apologies for adding "-otel" to the end of the file names earlier. I hope I didn't make too much trouble by doing that.

**Comment by jpipkin1** [comment]
@sumoanema, could you add the relevant Jira ticket number in the PR description?

Is this PR related to these tickets?
[SUMO-231512](https://sumologic.atlassian.net/browse/SUMO-231512)
[SUMO-231229](https://sumologic.atlassian.net/browse/SUMO-231229)

**Comment by kimsauce** [comment]
> @kimsauce, my apologies for adding "-otel" to the end of the file names earlier. I hope I didn't make too much trouble by doing that.

Oh, I didn't even notice. I thought it was part of the initial commit. I removed it because the URL already contains the otel keyword (`/opentelemetry-collector`). Long URLs, while they don't ding SEO, aren't a great user experience.

**Comment by sumoanema** [comment]
> @sumoanema, could you add the relevant Jira ticket number in the PR description?
> 
> Is this PR related to these tickets? [SUMO-231512](https://sumologic.atlassian.net/browse/SUMO-231512) [SUMO-231229](https://sumologic.atlassian.net/browse/SUMO-231229)

@jpipkin1 had added relevant jira to the PR

**Review by kimsauce — APPROVED** [review]

**Pull request merged**